cassovary-core/src/main/scala/com/twitter/cassovary/graph/MemoryMappedDirectedGraph.scala

```diff
@@ -51,48 +70,63 @@ class MemoryMappedDirectedGraph(file: File) extends DirectedGraph[Node] {
   private class MemoryMappedDirectedNode(override val id: Int) extends Node {
     val nodeOutboundOffset = outboundOffset(id)
     val nodeInboundOffset = inboundOffset(id)
-    override def outboundNodes(): Seq[Int] = new IndexedSeq[Int] {
+    def outboundNodes(): Seq[Int] = new IndexedSeq[Int] {
       val length: Int = outDegree(id)
       def apply(i: Int): Int =  data.getInt(nodeOutboundOffset + 4L * i)
-      override def toString(): String = "Memory Mapped IndexedSeq " + mkString("[", ",", "]")
     }
-    override def inboundNodes(): Seq[Int] = new IndexedSeq[Int] {
+    def inboundNodes(): Seq[Int] = new IndexedSeq[Int] {
       val length: Int = inDegree(id)
       def apply(i: Int): Int =  data.getInt(nodeInboundOffset + 4L * i)
-      override def toString(): String = "Memory Mapped IndexedSeq " + mkString("[", ",", "]")
     }
   }
 
-  override def getNodeById(id: Int): Option[Node] =
+  def getNodeById(id: Int): Option[Node] =
     if(0 <= id && id < nodeCount) {
       Some(new MemoryMappedDirectedNode(id))
     } else {
       None
     }
 
-  override def iterator: Iterator[Node] = (0 to nodeCount).iterator flatMap (i => getNodeById(i))
+  def iterator: Iterator[Node] = (0 to nodeCount).iterator flatMap (i => getNodeById(i))
 
-  override def edgeCount: Long = (outboundOffset(nodeCount) - outboundOffset(0)) / 4
+  lazy val edgeCount: Long = outboundOffset(nodeCount) - outboundOffset(0)
```

Comment:
I agree.  Using def gives us more flexibility, and the efficiency gain of lazy val (if there even is one) will be too small to notice.  I had it as a def, but for some reason Pankaj wanted it as a lazy val (I think for efficiency reasons), so I changed it before the cassovary merge. I'll make it a def.